addons/mail/models/mail_message.py

```diff
@@ -1007,11 +1007,16 @@ def _message_format(self, fnames):
                     })
 
             if message_sudo.model and message_sudo.res_id:
-                record_name = self.env[message_sudo.model] \
+                record = self.env[message_sudo.model] \
```

Comment:
If you keep `sudo()` at this step, please name it `record_sudo` to avoid mistakes.